joss-paper/paper.md

```diff
@@ -0,0 +1,159 @@
+    index: 3
+  - name: Leibniz Universität Hannover, D-30167 Hannover, Germany
+    index: 4
+date: 08 January 2021
+bibliography: paper.bib
+---
+
+# Summary
+
+Gravitational waves in the sensitivity band of ground-based detectors
+can be emitted by a number of astrophysical sources,
+including not only binary coalescences, but also individual spinning neutron stars.
+The most promising signals from such sources,
+although as of 2020 not yet detected,
+are the long-lasting, quasi-monochromatic 'Continuous Waves' (CWs).
+Many search methods have been developed and applied on
+LIGO [@TheLIGOScientific:2014jea]
+and Virgo [@TheVirgo:2014hva] data.
+See @Prix:2009oha, @Riles:2017evm, and @Sieniawska:2019hmd for reviews of the field.
+
+The `PyFstat` package provides a convenient interface
+to perform a range of CW data analysis tasks.
+It revolves around the $\mathcal{F}$-statistic,
+first introduced by @Jaranowski:1998qm:
+a matched-filter detection statistic for CW signals
+described by a set of frequency evolution parameters
+and maximized over amplitude parameters.
+It has been one of the standard methods for LIGO-Virgo CW searches for two decades.
+`PyFstat` is built on top of established routines
+in `LALSuite` [@lalsuite]
+but enables a flexible approach to designing new search strategies.
+
+Classes for various search strategies and target signals
+are contained in three main submodules:
+
+- `core` : The basic wrappers to `LALSuite`'s $\mathcal{F}$-statistic algorithm.
+End-users should rarely need to access these directly.
+- `grid_based_searches` : Classes to search over regular parameter-space grids.
+- `mcmc_based_searches` : Classes to cover small parameter-space regions
+<!-- around search targets or promising signal candidates from wider searches ('followup' use case) -->
+through stochastic template placement with the `ptemcee` sampler [@Vousden:2015pte].
+
+Besides standard CWs from isolated neutron stars, `PyFstat` can also be used
+for CWs from sources in binary systems (including the additional orbital parameters),
+for CWs with a discontinuity at a pulsar glitch,
+and for CW-like long-duration transient signals e.g. from _after_ a pulsar glitch.
+Specialized versions of both grid-based and MCMC-based search classes
+are provided for these scenarios.
+Both fully-coherent and semi-coherent searches
+(where the data is split into several segments for efficiency)
+are covered,
+and an extension to the $\mathcal{F}$-statistic
+that is more robust against single-detector noise artifacts
+[@Keitel:2013wga]
+is also supported.
+However, `PyFstat`'s grid-based searches do not compete with the sophisticated
+grid setups and semi-coherent algorithms implemented in various `LALSuite` programs.
+As discussed below, the main scientific use cases for `PyFstat` at the time of publication
+are for the MCMC exploration of small parameter-space regions
+and for the long-duration transient case.
```

Comment:
```suggestion
As discussed below, the main scientific use cases for `PyFstat` at the time of publication
are for the MCMC exploration of small parameter-space regions
and for the long-duration transient case.
However, `PyFstat`'s grid-based searches do not compete with the sophisticated
grid setups and semi-coherent algorithms implemented in various `LALSuite` programs.
```
I should massage this a little bit more, but do you think we should invert them?

Reply:
I think I prefer the original order as "end with an upper not a downer". But the `However` at the start could be dropped and instead somehow make it clearer from the `As discussed below` that the point is `However that's fine, our focus is different anyway`?

Reply:
Now that I read it again, you are right it's too much of a downer.

What about getting rid of `However`? Also, we know `small != bad`, but we can use a nicer word?
```suggestion
Although `PyFstat`'s grid-based searches do not compete with the sophisticated
grid setups and semi-coherent algorithms implemented in various `LALSuite` programs
its main scientific use cases for  at the time of publication are for the MCMC exploration 
of interesting parameter-space regions and for the long-duration transient case, as 
discussed below.
```

Reply:
```suggestion
While `PyFstat`'s grid-based searches do not compete with the sophisticated
grid setups and semi-coherent algorithms implemented in various `LALSuite` programs,
its main scientific use cases so far are for the MCMC exploration
of interesting parameter-space regions
and for the long-duration transient case,
as discussed below.
```

Reply:
No, that seems to focus the `as discussed below` on the transients. But since the "statement of need" is an expected default feature of a JOSS paper, we can just drop that comment:
 ```suggestion
While `PyFstat`'s grid-based searches do not compete with the sophisticated
grid setups and semi-coherent algorithms implemented in various `LALSuite` programs,
its main scientific use cases so far are for the MCMC exploration
of interesting parameter-space regions
and for the long-duration transient case.
```

Reply:
:+1: Let's go with it